src/Features/Homescreen.php

```diff
@@ -74,18 +75,63 @@ public function add_user_data_fields( $user_data_fields ) {
+
+			wc_admin_register_page(
+				array(
+					'id'         => 'woocommerce-home',
+					'title'      => __( 'WooCommerce', 'woocommerce-admin' ),
+					'path'       => self::MENU_SLUG,
+					'position'   => 10,
+					'capability' => 'read',
+					'icon'       => $woocommerce_icon,
+				)
+			);
+			return;
+		}
+
 		wc_admin_register_page(
 			array(
 				'id'         => 'woocommerce-home',
 				'title'      => __( 'Home', 'woocommerce-admin' ),
 				'parent'     => 'woocommerce',
 				'path'       => self::MENU_SLUG,
 				'order'      => 0,
-				'capability' => 'manage_woocommerce',
+				'capability' => 'read',
 			)
 		);
 	}
 
+	/**
+	 * Check if the user can access the top-level WooCommerce item.
+	 */
+	public function is_admin_user() {
+		return current_user_can( 'edit_others_shop_orders' ) || current_user_can( 'manage_woocommerce' );
+	}
+
+	/**
+	 * Possibly remove the WooCommerce menu item if it was purely used to access wc-admin pages.
+	 */
+	public function possibly_remove_woocommerce_menu() {
+		global $menu;
+
+		if ( $this->is_admin_user() ) {
+			return;
+		}
+
+		foreach ( $menu as $key => $menu_item ) {
+			if ( 'wc-admin' !== $menu_item[2] || 'read' !== $menu_item[1] ) {
```

Comment:
Maybe use `self::MENU_SLUG` here instead of `wc-admin`?

Reply:
Good call.